File: src/main/java/alfheim/common/potion/PotionBerserk.java
package alfheim.common.potion;

import java.util.UUID;

import alfheim.common.core.util.AlfheimConfig;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.SharedMonsterAttributes;
import net.minecraft.entity.ai.attributes.AttributeModifier;
import net.minecraft.entity.ai.attributes.BaseAttributeMap;

public class PotionBerserk extends PotionAlfheim {
	
	public PotionBerserk() {
		super(AlfheimConfig.potionIDBerserk, "berserk", false, 0xAA1111);
	}
	
	@Override
	public void applyAttributesModifiersToEntity(EntityLivingBase target, BaseAttributeMap map, int mod) {
		super.applyAttributesModifiersToEntity(target, map, mod);
		AttributeModifier m = new AttributeModifier(UUID.fromString("23593E88-FFE3-1707-DE94-6BCD1756B35D"), "berserk", -0.2, 2);
		target.getEntityAttribute(SharedMonsterAttributes.maxHealth).removeModifier(m);
		target.getEntityAttribute(SharedMonsterAttributes.maxHealth).applyModifier(m);
		target.setHealth(Math.min(target.getHealth(), target.getMaxHealth()));
	}
	
	@Override
	public void removeAttributesModifiersFromEntity(EntityLivingBase target, BaseAttributeMap map, int mod) {
		super.removeAttributesModifiersFromEntity(target, map, mod);
		AttributeModifier m = target.getEntityAttribute(SharedMonsterAttributes.maxHealth).getModifier(UUID.fromString("23593E88-FFE3-1707-DE94-6BCD1756B35D"));
		if (m != null) target.getEntityAttribute(SharedMonsterAttributes.maxHealth).removeModifier(m);
	}
}

File: src/main/java/alfheim/common/potion/PotionBerserk.kt
package alfheim.common.potion

import alfheim.common.core.handler.AlfheimConfigHandler
import net.minecraft.entity.*
import net.minecraft.entity.ai.attributes.*
import java.util.*

class PotionBerserk: PotionAlfheim(AlfheimConfigHandler.potionIDBerserk, "berserk", false, 0xAA1111) {
	
	val uuid = UUID.fromString("23593E88-FFE3-1707-DE94-6BCD1756B35D")!!
	
	override fun applyAttributesModifiersToEntity(target: EntityLivingBase?, map: BaseAttributeMap, mod: Int) {
		super.applyAttributesModifiersToEntity(target, map, mod)
		val m = AttributeModifier(uuid, "berserk", -0.2, 2)
		target!!.getEntityAttribute(SharedMonsterAttributes.maxHealth).removeModifier(m)
		target.getEntityAttribute(SharedMonsterAttributes.maxHealth).applyModifier(m)
		target.health = target.health.coerceAtMost(target.maxHealth)
	}
	
	override fun removeAttributesModifiersFromEntity(target: EntityLivingBase?, map: BaseAttributeMap, mod: Int) {
		super.removeAttributesModifiersFromEntity(target, map, mod)
		val m = target!!.getEntityAttribute(SharedMonsterAttributes.maxHealth).getModifier(uuid)
		if (m != null) target.getEntityAttribute(SharedMonsterAttributes.maxHealth).removeModifier(m)
	}
}
